Assets/MRTK/SDK/Editor/Inspectors/UX/StateVisualizer/StateVisualizerInspector.cs

```diff
@@ -0,0 +1,402 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+using Microsoft.MixedReality.Toolkit.UI.Interaction;
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    /// <summary>
+    /// Custom inspector for the StateVisualizer component
+    /// </summary>
+    [CanEditMultipleObjects]
+    [CustomEditor(typeof(StateVisualizer))]
+    public class StateVisualizerInspector : UnityEditor.Editor
+    {
+        private StateVisualizer instance;
+        private SerializedProperty interactiveElement;
+        private SerializedProperty animator;
+        private SerializedProperty stateContainers;
+
+        private bool inPlayMode;
+
+        private static GUIContent RemoveButtonLabel;
+        private static GUIContent AddButtonLabel;
+
+        private AnimatableProperty animatableProperty;
+
+        protected virtual void OnEnable()
+        {
+            instance = (StateVisualizer)target;
+            interactiveElement = serializedObject.FindProperty("interactiveElement");
+            animator = serializedObject.FindProperty("animator");
+
+            stateContainers = serializedObject.FindProperty("stateContainers");
+
+            RemoveButtonLabel = new GUIContent(InspectorUIUtility.Minus, "Remove");
+            AddButtonLabel = new GUIContent(InspectorUIUtility.Plus, "Add");
+        }
+
+        public override void OnInspectorGUI()
+        {
+            inPlayMode = EditorApplication.isPlaying || EditorApplication.isPaused;
+
+            serializedObject.Update();
+
+            RenderExperimentalWarning();
+
+            RenderInitialProperties();
+
+            if (instance.GetComponent<Animator>().runtimeAnimatorController != null)
+            {
+                RenderStateContainers();
+            }
+
+            RenderEndingButtons();
+
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        private void RenderExperimentalWarning()
```

Comment:
Can you re-use the ExperimentalDrawer class rather that re-implementing the HelpBox rendering here? Would be good to reduce code duplication so that we don't need to change this message below in multiple places if we decide to change the message in the future.